docs/UserGuide.md

```diff
@@ -162,6 +162,44 @@ The list entry of the user with <id> will not have the tag anymore.
 
 ![delete-tag](images/user-guide/delete-tag_mock_output.png)
 
+### Filtering contacts : filter
+
+Filters the contacts.
+
+Format: `filter <component>[.<modifier>]/<value> ...`
+
+`component` is one of `n`,`p`,`e`, or `a` corresponding to the values in add:
+name, phone, email and address respectively.
+
+There can be duplicate components, if there are multiple components, the
+contacts that match any of the components are shown.
+
+In order to filter with values that must all match, the only way to do so right
+now is to run multiple sequential filter commands.
+
+`modifier` is to specify how the filtering should be done, it is optional and
+defaults to `has`. The components are
+- `has`: value has to match part of the component
+- `hasnt`: negation of has
+- `is`: value has to match the component exactly
+- `isnt`: negation of is
+- `word`: value has to match a distinct word in the component, a word is any
+sequence of letters and numbers surrounded by spaces
+- `noword`: negation of word
+- `none`: the component is empty
+- `any`: the component is not empty
+
+Example
+```
+filter a/queenstown
+> The default modifier is has, so this lists every contact with an address that
+> has queenstown in it.
+
+filter p.is/12345678
```

Comment:
Can you make the command "p is/12345678" in filtering by phone which makes it more user-friendly

Reply:
Hmm I don't see how that'd make it any more user friendly. I think it might moreso confuse the user that there is some is component.

On the implementation side of things, it'd be really difficult to parse something like
```
filter a has/something p is/12345678
```
because in our current implementation, `has` will have the component `something p`.

For parsing `p.is` style syntax would only need to extend the current prefix parsing functionality.